demonstrations/tutorial_intro_qsvt.py

```diff
@@ -0,0 +1,294 @@
+
+Transforming scalars encoded in matrices
+-----------------------------------------
+My personal perspective on QSVT is that it is really a result in linear algebra that tells us how to
+transform matrices encoded in larger unitary matrices.
+
+Let's start with the simplest example:
+we encode a scalar :math:`a` inside a 2x2 unitary :math:`U(a)`. By encoding we mean that the
+matrix form of the unitary depends explicitly on :math:`a`. This encoding can be achieved in
+multiple ways, for example:
+
+.. math:: U(a) = \begin{pmatrix} a & \sqrt{1-a^2}\\
+    \sqrt{1-a^2} & -a
+    \end{pmatrix}.
+
+The parameter :math:`a` must lie between -1 and 1 to ensure the operator is unitary, but this is just a matter of rescaling.
+
+We now ask the crucial question that will get everything started:
+what happens if we repeatedly alternate multiplication of this unitary by some other unitary? 🤔
+There are multiple choices for the "other unitary", for example
+
+.. math:: S(\phi) = \begin{pmatrix} e^{i\phi} & 0\\
+    0 & e^{-i\phi}
+    \end{pmatrix},
+
+which has the advantage of being diagonal. This is known as the **signal-processing** operator.
+
+The answer to our crucial question is encapsulated in a result known as **quantum signal processing** (QSP).
+If we alternate products of :math:`U(a)` and :math:`S(\phi)`, keeping :math:`a` fixed and varying :math:`\phi`,
+the top-left corner of the resulting matrix is a polynomial transformation of :math:`a`. Mathematically,
+
+.. math:: S(\phi_0)\prod_{k=1}^d W(a) S(\phi_k) = \begin{pmatrix}
+    P(a) & *\\
+    * & *
+    \end{pmatrix}.
+
+The asterisk :math:`*` is used to indicate that we are not interested in these entries.
+The complex polynomial :math:`P(a)` has degree at most :math:`d`, determined by the number of angles,
+and its particular form depends on the choice of angles.
+
+The main quantum signal processing theorem states that there exist a sequence of :math:`d+1` angles
+that can implement *any* complex polynomial of degree :math:`d`.
+Finding the desired angles can be done efficiently in practice, but identifying the best
+methods is an active area of research.
+
+For now, let's look at a simple example of how quantum signal processing can be implemented using
+PennyLane. We aim to perform a transformation by the Legendre polynomial
+:math:`(5 x^3 - 3x)/2`, for which we use pre-computed optimal angles.
+As you will soon learn, QSP can be viewed as a special case of QSVT. We thus use the :class:`~pennylane.qsvt`
+operation to construct the output matrix and compare the resulting transformation to
+the target polynomial.
+
+"""
+
+import pennylane as qml
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def target_poly(a):
+    return 0.5 * (5 * a**3 - 3 * a)
+
+
+# pre-optimized angles
+angles = [-0.20409113, -0.91173829, 0.91173829, 0.20409113]
+
+
+def qsvt_output(a):
+    # output matrix
+    out = qml.matrix(qml.qsvt(a, angles, wires=[0]))
+
+    return out[0, 0]  # top-left entry
```

Comment:
Just a suggestion to keep things more compact: 

```suggestion
def qsvt_output(a):
    # output matrix
    out = qml.matrix(qml.qsvt(a, angles, wires=[0]))
    return out[0, 0]  # top-left entry
```